src/nacl/bindings/crypto_generichash.py

```diff
@@ -122,6 +122,26 @@ def generichash_blake2b_salt_personal(data,
     return ffi.buffer(digest, digest_size)[:]
 
 
+class Blake2State(object):
+    """
+    Python-level wrapper for the crypto_generichash_blake2b state buffer
+    """
+    __slots__ = ['_statebuf', 'digest_size']
+
+    def __init__(self, digest_size):
+        self._statebuf = ffi.new("unsigned char[]",
+                                 crypto_generichash_STATEBYTES)
+        self.digest_size = digest_size
+
+    def __copy__(self):
+        _st = self.__class__(self.digest_size)
+        ffi.memmove(_st._statebuf,
+                    self._statebuf, crypto_generichash_STATEBYTES)
+        return _st
+
+    copy = __copy__
```

Comment:
Not sure how I feel about this. Implementing `__copy__` implies using `copy.copy` is what you want to support, right (or really `copy.deepcopy` based on my comment above). So is there a reason to alias the method in this fashion rather than `from copy import deepcopy; newst = deepcopy(stateobj)`?

Reply:
The aliasing was not really needed. Removed.